File: Api/src/main/java/net/binggl/mydms/features/startpage/UserInfo.java
package net.binggl.mydms.features.startpage;

import net.binggl.mydms.features.security.models.User;

public class UserInfo extends User {

	private static final long serialVersionUID = 1L;

	public UserInfo() {
		super("","","","",null);
	}
	
	public UserInfo(User user) {
		super(user.getUserId(),user.getUserName(),user.getDisplayName(),user.getEmail(),null);
	}
}

File: Api/src/main/kotlin/net/binggl/mydms/features/appinfo/models/UserInfo.kt
package net.binggl.mydms.features.appinfo.models

import net.binggl.mydms.shared.models.Claim
import net.binggl.mydms.shared.models.User
import java.io.Serializable

class UserInfo(val userId: String, val userName: String, val displayName: String, val claims: List<Claim>?) : Serializable {

    constructor(user: User) : this(user.userId, user.userName, user.displayName, user.claims){
    }

    companion object {
        private val serialVersionUID = 1L
    }
}